File: src/main/java/com/inanc/smartcommit/LocalStorage.java
package com.inanc.smartcommit;

import java.util.prefs.Preferences;

public class LocalStorage {

    private Preferences prefs = Preferences.userNodeForPackage(LocalStorage.class);

    // Save a value
    public void saveValue(String key, String value) {
        prefs.put(key, value);
    }

    public void removeValue(String key){
        prefs.remove(key);
    }

    // Load a value
    public String loadValue(String key) {
        return prefs.get(key, null);  // null is the default value if key is not found
    }

    public void saveTerms(String key, Boolean value) {
        prefs.putBoolean(key, value);
    }

    public Boolean loadValueAcceptTerms(String acceptTerms) {
      return   prefs.getBoolean(acceptTerms,false);
    }
}

File: src/main/java/com/inanc/smartcommit/LocalStorage.kt
package com.inanc.smartcommit

import java.util.prefs.Preferences

class LocalStorage {

    private val prefs: Preferences = Preferences.userNodeForPackage(LocalStorage::class.java)

    // Save a value
    fun saveValue(key: String, value: String) {
        prefs.put(key, value)
    }

    fun removeValue(key: String) {
        prefs.remove(key)
    }

    // Load a value
    fun loadValue(key: String): String? {
        return prefs.get(key, null) // null is the default value if key is not found
    }

    fun saveTerms(key: String, value: Boolean) {
        prefs.putBoolean(key, value)
    }

    fun loadValueAcceptTerms(acceptTerms: String): Boolean {
        return prefs.getBoolean(acceptTerms, false)
    }
}
